packages/grant-explorer/src/features/round/ViewCartPage/MRCProgressModalBody.tsx

```diff
@@ -0,0 +1,177 @@
+                />
+              ) : step.status === ProgressStatus.IN_PROGRESS ? (
+                <MRCModalStep
+                  step={step}
+                  icon={
+                    <span className="relative z-10 w-8 h-8 flex items-center justify-center bg-white border-2 border-violet-500 rounded-full">
+                      <span
+                        className="h-2.5 w-2.5 bg-violet-500 rounded-full animate-pulse-scale"
+                        data-testid={`${step.name}-current-icon`}
+                      />
+                    </span>
+                  }
+                  line={
+                    <div
+                      className="-ml-px absolute mt-0.5 top-4 left-4 w-0.5 h-full bg-grey-200"
+                      aria-hidden="true"
+                    />
+                  }
+                  nameColor="text-violet-500"
+                  isLastStep={stepIdx === steps.length - 1}
+                />
+              ) : step.status === ProgressStatus.IS_ERROR ? (
+                <MRCModalStep
+                  step={step}
+                  icon={
+                    <span className="relative z-10 w-8 h-8 flex items-center justify-center border-2 bg-white border-pink-500 rounded-full">
+                      <XMarkIcon
+                        className="w-5 h-5 text-pink-500"
+                        data-testid={`${step.name}-error-icon`}
+                      />
+                    </span>
+                  }
+                  line={
+                    <div
+                      className="-ml-px absolute mt-0.5 top-4 left-4 w-0.5 h-full bg-grey-300"
+                      aria-hidden="true"
+                    />
+                  }
+                  isLastStep={stepIdx === steps.length - 1}
+                  nameColor="text-grey-500"
+                />
+              ) : step.status === ProgressStatus.NOT_STARTED ? (
+                <MRCModalStep
+                  step={step}
+                  icon={
+                    <span
+                      className="relative z-10 w-8 h-8 flex items-center justify-center bg-white border-2 rounded-full border-grey-400"
+                      data-testid={`${step.name}-upcoming-icon`}
+                    ></span>
+                  }
+                  line={
+                    <div
+                      className="-ml-px absolute mt-0.5 top-4 left-4 w-0.5 h-full bg-grey-300"
+                      aria-hidden="true"
+                    />
+                  }
+                  isLastStep={stepIdx === steps.length - 1}
+                  nameColor="text-grey-400"
+                />
+              ) : (
+                <></>
+              )}
+            </li>
+          ))}
+        </ol>
+      </nav>
+    </>
+  );
+}
+
+function MRCModalStep(props: {
+  step: Step;
+  icon: JSX.Element;
+  line: JSX.Element;
+  nameColor: string;
+  descriptionColor?: string;
+  isAriaHidden?: boolean;
+  isAriaCurrent?: boolean;
```

Comment:
It should be possible to do this without introducing ad-hoc props with something like:

```tsx
interface FooProps {
  bar: number
}

function Foo({bar, ...divProps}: FooProps & Pick<HTMLDivElement, 'ariaHidden' | 'ariaCurrent'>) { 
  return <div {...divProps}>{bar}</div>
}
```